chromium_src/ui/views/widget/native_widget_mac.h

```diff
@@ -8,8 +8,19 @@
 
 #include "ui/views/widget/native_widget_private.h"
 
-#define SetWindowIcons                    \
-  SetWindowTitleVisibility(bool visible); \
+#define SetWindowIcons                                      \
+  SetWindowTitleVisibility(bool visible);                   \
+  bool has_overridden_window_title_visibility() const {     \
+    return overridden_window_title_visibility_.has_value(); \
+  }                                                         \
+  bool overridden_window_title_visibility() const {         \
+    return overridden_window_title_visibility_.value();     \
```

Comment:
Should we do a `.value_or(false)` here? It's not obvious that the method will crash if you don't check `has_overridden_window_title_visibility` first.

Reply:
> It's not obvious that the method will crash

I thought it'd crash, but it seems it's close to undefined behavior. I think it's better to let it crash to detect problems for some cases, and this is the case. And also I intentionally didn't use `.value_or()` here, because when there's no overridden visibility, we should use WidgetDelegate::ShouldShowWindowTitle() to know about the default visibility. 

Now I find it that `DCHECK(has_overriden_visibility()) << "Use WidgetDelegate::ShouldShowWindowTitle()"` is more explicit and better, thanks to your comment.